File: logica-de-programacao/secao-11/exercicios/diagonal_negativos.java
package exercicios;

import java.util.*;

public class diagonal_negativos {
    public static void main(String[] args) {
        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);

        int negativo = 0;

        System.out.print("Qual a ordem da matriz? ");
        int n = sc.nextInt();

        int mat[][] = new int[n][n];

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                System.out.print("Elemento[" + i + "," + j + "]: ");
                mat[i][j] = sc.nextInt();

                if (mat[i][j] < 0) {
                    negativo = negativo + 1;
                }
            }
        }

        System.out.println("DIAGONAL PRINCIPAL: ");
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (i == j) {
                    System.out.print(mat[i][j] + " "); 
                }
            }
        }
        System.out.println();

        System.out.println("QUANTIDADE DE NEGATIVOS = " + negativo);

        sc.close();
    }
}

File: logica-de-programacao/secao-13/exercicios/diagonal_negativos.py
negativos = 0

n = int(input("Qual a ordem da matriz? "))

mat = [[0 for x in range(n)] for x in range(n)]

for i in range(0, n):
    for j in range(0,n):
        mat[i][j] = int(input(f"Elemento [{i},{j}]: "))
        if mat[i][j] < 0:
            negativos += 1

print("Diagonal principal:")

for i in range(0, n):
    for j in range(0,n):
        if i == j:
            print(f"{mat[i][j]} ", end="")

print()
print(f"Quantidade de negativos = {negativos}")


